connection.go

```diff
@@ -581,43 +591,63 @@ func pack(h *smallWBuf, enc *encoder, reqid uint32,
 	return
 }
 
-func (conn *Connection) writeAuthRequest(w *bufio.Writer, scramble []byte) (err error) {
+func (conn *Connection) writeRequest(w *bufio.Writer, req Request,
+	reqName string) (err error) {
 	var packet smallWBuf
-	req := newAuthRequest(conn.opts.User, string(scramble))
 	err = pack(&packet, newEncoder(&packet), 0, req, ignoreStreamId, conn.Schema)
 
 	if err != nil {
-		return errors.New("auth: pack error " + err.Error())
+		return errors.New(reqName + ": pack error " + err.Error())
 	}
 	if err := write(w, packet.b); err != nil {
-		return errors.New("auth: write error " + err.Error())
+		return errors.New(reqName + ": write error " + err.Error())
 	}
 	if err = w.Flush(); err != nil {
-		return errors.New("auth: flush error " + err.Error())
+		return errors.New(reqName + ": flush error " + err.Error())
 	}
 	return
 }
 
-func (conn *Connection) readAuthResponse(r io.Reader) (err error) {
+func (conn *Connection) writeAuthRequest(w *bufio.Writer, scramble []byte) (err error) {
+	req := newAuthRequest(conn.opts.User, string(scramble))
+	return conn.writeRequest(w, req, "auth")
+}
+
+func (conn *Connection) writeProtocolInfoRequest(w *bufio.Writer, version ProtocolVersion,
+	features []Feature) (err error) {
+	req := newProtocolInfoRequest(version, features)
+	return conn.writeRequest(w, req, "iproto id")
+}
+
+func (conn *Connection) readResponse(r io.Reader, reqName string) (resp Response, err error) {
```

Comment:
Same as for `writeRequest` (error wrapping).

Reply:
Fixed